### PR TITLE
Update app disconnect copy in connected-apps

### PR DESCRIPTION
**Pull request by LeKeve**
## Description

This PR updates the copy in the alert for disconnecting third party applications. The privacy policy url is fetched from the Apps API service. 

## Tickets

[4003](https://vajira.max.gov/browse/API-4003)

## Testing done

- local testing with demo apps in profile
- updated unit tests

## Screenshots

Screen Shot 2021-01-14 at 11 50 02 AM

## Acceptance criteria
- [x] Updated copy in third party app disconnect alert

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx [hunk 1]
```diff
@@ -5,13 +5,37 @@ import { focusElement } from 'platform/utilities/ui';
...
     focusElement('[data-focus-target]');
   }, []);
-
-  const { id, title, dismissAlert } = props;
-  const alertMessage = `We’ve disconnected ${title}. This app can’t access any new information from your VA.gov profile. Some apps may still store information you’ve already shared. If you’d like to ask the app to delete any stored information, contact the app’s support.`;
+  const privacyLink = (
+    <a
+      href={`${privacyPolicies[title]}`}
+      target="_blank"
+      rel="noopener noreferrer"
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx:21]: Might be nice to add an `aria-label` that adds the name of the app. But that might be overkill. Perhaps @joshkimux could weigh in on that.

File: src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx [hunk 2]
```diff
@@ -5,13 +5,37 @@ import { focusElement } from 'platform/utilities/ui';
...
+}) => {
   useEffect(() => {
     focusElement('[data-focus-target]');
   }, []);
-
-  const { id, title, dismissAlert } = props;
-  const alertMessage = `We’ve disconnected ${title}. This app can’t access any new information from your VA.gov profile. Some apps may still store information you’ve already shared. If you’d like to ask the app to delete any stored information, contact the app’s support.`;
+  const privacyLink = (
+    <a
+      href={`${privacyPolicies[title]}`}
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx:19]: It seems like it's possible for this `href` to be undefined if the call to look up the privacy policy URL is still pending or failed.

**Review by erikphansen — COMMENTED** [review]
I'd push really hard for the `GET profile/connected_applications` response to be modified so that each app's privacy policy URL was part of the attributes. That would make the FE implementation a _whole_ lot more straightforward. It feels a little wrong to make an additional API request when you start the delete process _just_ to get a URL. What do you think?

File: src/applications/personalization/profile/components/connected-apps/actions/index.js [hunk 3]
```diff
@@ -93,3 +98,15 @@ export function dismissDeletedAppAlert(appId) {
   return async dispatch =>
     dispatch({ type: DELETED_APP_ALERT_DISMISSED, appId });
 }
+
+export function getPrivacyPolicy(appTitle) {
+  return async dispatch => {
+    await apiRequest(`${appsUrl}/${appTitle}`, { apiVersion: 'services' })
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/connected-apps/actions/index.js:104]: Does this API actually work from the local env?

**Comment by LeKeve** [comment]
> I'd push really hard for the `GET profile/connected_applications` response to be modified so that each app's privacy policy URL was part of the attributes. That would make the FE implementation a _whole_ lot more straightforward. It feels a little wrong to make an additional API request when you start the delete process _just_ to get a URL. What do you think?

@erikphansen I think that approach makes a lot more sense. I initially took this route because the privacy policy urls aren't readily available through Okta. After speaking with my team it looks like we can update the `GET profile/connected_applications` response to include privacy policy links.

File: src/applications/personalization/profile/components/connected-apps/actions/index.js (hunk 3 above)
Comment (LeKeve) [reply src/applications/personalization/profile/components/connected-apps/actions/index.js:104]: Yes. The apps api service is not at the root api route and is versioned separately. The endpoint structure looks like this: `/services/apps/v0/directory/`, so here I'm overwriting the api version.

File: src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx (hunk 1 above)
Comment (LeKeve) [reply src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx:21]: Thanks for catching that! I agree that it should be included to meet the 508 standards. Will include in update.

File: src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx (hunk 2 above)
Comment (LeKeve) [reply src/applications/personalization/profile/components/connected-apps/AppDeletedAlert.jsx:19]: Will be resolved by updates to the `GET profile/connected_applications` response.

**Comment by erikphansen** [comment]
That is _fantastic_ news, @LeKeve! Thanks for checking on that.

**Comment by LeKeve** [comment]
Closing this PR for now. Will be updating the connected apps response to include a privacy url property and opening a new PR at a later time.

**Pull request closed without merging**